### ISSUE TITLE
URLLib3Session send has incorrect type hints

### ISSUE THREAD

**Issue by SteggyLeggy** [youtype/botocore-stubs#4]
I believe the send function of has been updated to have some incorrect type hints.

```
def send(self, request: AWSRequest) -> None: ...
```

We've started getting a mypy error
```
error: Argument 1 to "send" of "URLLib3Session" has incompatible type "AWSPreparedRequest"; expected "AWSRequest"  [arg-type]
```
It's tricky to tell from the botocore source if the intention of the send function of URLLib3Session was intended to be a AWSPreparedRequest or AWSRequest (or perhaps maybe both?) But we have been passing in a prepared request for some time without any issue. And from what I can see of the source code, for that function, it doesn't use any part of the `request` that isn't in both AWSRequest and AWSPreparedRequest.

Also I'm a lot more sure of the return type being incorrect, as the send function clearly does this
```
    http_response = botocore.awsrequest.AWSResponse(
        request.url,
        urllib_response.status,
        urllib_response.headers,
        urllib_response,
    )
            
    # ... trimmed for brevity 
            
    return http_response
```

So I'm kind of inclined to believe the type hinting for send should be something like this...
```
def send(self, request: Union[AWSRequest, AWSPreparedRequest]) -> AWSResponse: ...
```

**Comment by vemel** [comment]
Thank you for the report!

Fixed and released `botocore-stubs 1.31.12.post1`. Please test once you have time and let me know if it works as it should.

**Comment by SteggyLeggy** [comment]
Amazing turn around, thank you @vemel yes that's fixed the issue.
